ingest/ingest_files.py

```diff
@@ -0,0 +1,120 @@
+"""Module for text, CSV, and TSV file types used in SCP ingest
+
+DESCRIPTION
+Module provides extract capabilities for text, CSV, and TSV file types
+
+"""
+import csv
+import mimetypes
+import os
+import re
+from itertools import islice
+
+
+class IngestFiles:
+    def __init__(self, file_path, allowed_file_types):
+        if not os.path.exists(file_path):
+            raise IOError(f"File '{file_path}' not found")
+        self.allowed_file_types = allowed_file_types
+        self.file_type, self.file = self.open_file(file_path)
+        # Keeps tracks of lines parsed
+        self.amount_of_lines = 0
+
+    def open_file(self, file_path):
+        """ Opens txt, csv, or tsv formmatted files"""
+        open_file = open(file_path, encoding='utf-8-sig')
+        file_connections = {
+            # Remove BOM with encoding='utf-8-sig'
+            'text/csv': self.open_csv(open_file),
+            'text/plain': open_file,
+            'text/tab-separated-values': self.open_tsv(open_file),
+        }
+        # Check file type
+        file_type = self.get_file_type(file_path)[0]
+        # See if file type is allowed
+        if file_type in self.allowed_file_types:
+            # Return file object and type
+            return file_type, file_connections.get(file_type)
+        else:
+            raise ValueError('Unsupported file format')
+
+    # Inherited function
+    def extract(self):
+        """ Calls extract function for txt, csv, or tsv formmatted files to
+            retrieve all contents from file.
+        """
+
+        file_type_extract_fns = {
+            'text/csv': self.extract_csv_or_tsv,
+            'text/plain': self.extract_txt,
+            'text/tab-separated-values': self.extract_csv_or_tsv,
+        }
+        return file_type_extract_fns.get(self.file_type)()
+
+    def split_line(self, line):
+        """Splits lines on spaces, tabs, or commas"""
+        return re.findall(r'[^,\s\t]+', line)
```

Comment:
I'm concerned about splitting on tab, comma, and space here.  We will have instances (especially with cell metadata files) where there may be a value that is a short phrase that we will want to preserve, and if we split on space then this will break.